tests/common/test_cgroupstelemetry.py

```diff
@@ -63,34 +71,57 @@ def consume_memory():
     return waste
 
 
-def make_root_cgroup():
-    cpu_controller_id = CGroupsApi._get_controller_id('cpu')
-    current_process_cpu_cgroup_path = CGroupsApi._get_current_process_cgroup_relative_path(cpu_controller_id)
+def make_new_cgroup(name="test-cgroup"):
+    return CGroupConfigurator.get_instance().create_extension_cgroups(name)
 
-    memory_controller_id = CGroupsApi._get_controller_id('memory')
-    current_process_memory_cgroup_path = CGroupsApi._get_current_process_cgroup_relative_path(memory_controller_id)
 
-    return [CGroup.create(current_process_cpu_cgroup_path, "cpu", "test-cgroup"),
-            CGroup.create(current_process_memory_cgroup_path, "memory", "test-cgroup")]
+class TestCGroupsTelemetry(AgentTestCase):
+    @classmethod
+    def tearDownClass(cls):
+        pass
 
+    def setUp(self):
+        prefix = "{0}_".format(self.__class__.__name__)
 
-def make_self_cgroup(name="test-cgroup"):
-    current_process_cgroup_suffix = ""
-    cgroups = []
-    for controller in ["cpu", "memory"]:
-        try:
-            controller_id = CGroupsApi._get_controller_id(controller)
-            current_process_cgroup_suffix = CGroupsApi._get_current_process_cgroup_relative_path(controller_id)
-        except Exception:
-            pass
+        self.temp_dir = tempfile.mkdtemp(prefix=prefix)
 
-        cgroups.append(CGroup.create(os.path.join(CGROUPS_FILE_SYSTEM_ROOT, controller,
-                                                  current_process_cgroup_suffix), controller, name))
+        ext_handler_properties = ExtHandlerProperties()
+        ext_handler_properties.version = "1.2.3"
+        self.ext_handler = ExtHandler(name='foo')
+        self.ext_handler.properties = ext_handler_properties
+        self.ext_handler_instance = ExtHandlerInstance(ext_handler=self.ext_handler, protocol=None)
 
-    return cgroups
+        self.mock_get_base_dir = patch("azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_base_dir",
+                                       lambda *_: self.temp_dir)
+        self.mock_get_base_dir.start()
 
+        self.log_dir = os.path.join(self.temp_dir, "log")
+        self.mock_get_log_dir = patch("azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_log_dir",
+                                      lambda *_: self.log_dir)
+        self.mock_get_log_dir.start()
+
+    def tearDown(self):
+        if self.temp_dir is not None:
+            shutil.rmtree(self.temp_dir)
+
+    def _create_script(self, file_name, contents):
```

Comment:
we should not duplicate this code... if needed across multiple tests let's move it to a common module